### PR TITLE
refactor: define typescript types for CheckCancel Transaction Model 

### PR DESCRIPTION
**Pull request by mukulljangid**
## High Level Overview of Change
Defines typescript types for CheckCancel transaction.

### Context of Change
CheckCancel is an extension of CommonFields, and is exported in a type union with all other transaction types defined as Transaction.

We also will provide an optional verifyCheckCancel function for runtime form checking of CheckCancel transactions, should the users want to verify transaction form at runtime.

### Type of Change

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [x] Refactor (non-breaking change that only restructures code)
- [ ] Tests (You added tests for code that already exists, or your new feature included in this PR)
- [ ] Documentation Updates
- [ ] Release

## Test Plan
Adds tests for `verifyCheckCancel`, which provides runtime form verification for `CheckCancel` transactions

### CODE REVIEW

**Review by natenichols — APPROVED** [review]
You'll have to fix the merge conflicts, but once that is done this LGTM 👍

File: src/models/transactions/transaction.ts [hunk 1]
```diff
@@ -1,20 +1,19 @@
...
 //   | CheckCash
-      CheckCreate
+//   | CheckCreate
 //   | DepositPreauth
 //   | EscrowCancel
 //   | EscrowCreate
 //   | EscrowFinish
 //   | OfferCancel
-    |  OfferCreate
+     | OfferCreate
```
Comment (mvadari) [inline src/models/transactions/transaction.ts:16]: merge issues here - fix this

File: src/models/transactions/index.ts [hunk 2]
```diff
@@ -1,3 +1,3 @@
 export * from './transaction'
 export * from './offerCreate'
-export * from './checkCreate'
+export * from './checkCancel'
```
Comment (mvadari) [inline src/models/transactions/index.ts:3]: ditto

**Review by mvadari — APPROVED** [review]
lgtm - just a couple merge issues to fix

**Review by mvadari — APPROVED** [review]

**Pull request merged**